### PR TITLE
ci: update dependabot workflow 

### PR DESCRIPTION
**Pull request by maschad**
This fixes an issue with https://github.com/FuelLabs/fuels-ts/pull/2102 where the changeset update workflow [didn't install](https://github.com/FuelLabs/fuels-ts/actions/runs/8745556744/job/24000757173?pr=2097) `pnpm` to run the changeset script.

### CODE REVIEW

File: .github/workflows/dependabot-changeset.yaml [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
-      - master
+  repository_dispatch:
+    types: [create-changeset]
 
 jobs:
   create-changeset:
     name: Create Changeset
     runs-on: ubuntu-latest
-    if: github.actor == 'dependabot[bot]'
+    ## Comment out the following line to enable the workflow
```
Comment (maschad) [inline .github/workflows/dependabot-changeset.yaml:11]: Remember to uncomment

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -146,3 +146,37 @@ jobs:
           comment_tag: diff
           mode: recreate
           create_if_not_exists: true
+  create-changeset:
+    needs: [test]
+    name: Create Changeset
+    runs-on: ubuntu-latest
+    ## Comment out the following line to enable the workflow
+    #  github.actor == 'dependabot[bot]'
```
Comment (maschad) [inline .github/workflows/test.yaml:154]: Remember to uncomment

File: .changeset/fast-berries-sneeze.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+---
+
+Merge b46be6c49c490b2116ebed84ec254e0acafb8199 into bbd31a192e4de862c653fb1d30e813773716df43
```
Comment (maschad) [inline .changeset/fast-berries-sneeze.md:4]: Looks like we are [in the same boat](https://github.com/FuelLabs/fuels-ts/actions/runs/8879891098/job/24378703778?pr=2191#step:5:20) @nedsalk 😓

**Comment by maschad** [comment]
This is [ready for review](https://github.com/FuelLabs/fuels-ts/pull/2131/files#diff-2f53566278cf7f2f103221ea8cbc7cf52d243f4f63f6e41358a15ea61b74a441R4) now

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -147,3 +147,38 @@ jobs:
...
+          ref: ${{ github.event.client_payload.ref }}
+
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v2.1.0
+        with:
+          version: 9.0.5
+          run_install: true
+
+      - name: Install jq
+        run: sudo apt-get install -y jq
```
Comment (danielbate) [inline .github/workflows/test.yaml:169]: What is `jq`?

Comment (arboleya) [reply .github/workflows/test.yaml:169]: https://jqlang.github.io/jq/

**Review by nedsalk — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Pull request merged**